src/local_scheduler/local_scheduler.h

```diff
@@ -92,20 +92,46 @@ void kill_worker(LocalSchedulerClient *worker, bool wait);
 void start_worker(LocalSchedulerState *state, ActorID actor_id);
 
 /**
- * Update our accounting for the current resources being used, according to
- * some task that is starting or finishing execution.
+ * Check if a certain quantity of dynamic resources are available.
+ *
+ * @param state The state of the local scheduler.
+ * @param num_cpus Check if this many CPUs are available.
+ * @param num_gpus Check if this many GPUs are available.
+ * @return True if there are enough CPUs and GPUs and false otherwise.
+ */
+bool check_dynamic_resources(LocalSchedulerState *state,
+                             double num_cpus,
+                             double num_gpus);
+
+/**
+ * Acquire resources (CPUs and GPUs) for a worker. If the worker already has
+ * some resources allocated for it, this will acquire additional resources.
```

Comment:
Hmm, this isn't actually true for GPUs, is it? The checks in `acquire_resources` don't seem to allow you to acquire additional GPUs.